dwa_local_planner/src/dwa_planner_ros.cpp

```diff
@@ -313,13 +315,15 @@ namespace dwa_local_planner {
       ROS_DEBUG_NAMED("dwa_local_planner", "Slowing down from (%.2f, %.2f, %.2f) to (%.2f, %.2f, %.2f)",
                       vx, vy, vt, cmd_vel.twist.linear.x, cmd_vel.twist.linear.y, cmd_vel.twist.angular.z);
 
-      message = "The dwa local planner failed to find a valid plan, cost functions discarded all candidates. This can mean there is an obstacle too close to the robot";
+      message = "The dwa local planner failed with an error: " + std::to_string(outcome);
       ROS_DEBUG_STREAM_NAMED("dwa_local_planner", message);  //// TODO why is this DEBUG?
       local_plan.clear();
       publishLocalPlan(local_plan);
-      return mbf_msgs::ExePathResult::NO_VALID_CMD;
+      return outcome;
     }
 
+    assert(path.cost_ >= 0);
```

Comment:
S: Maybe remove this? Asserts don't run on release mode anyway so we'll never trigger it

Reply:
yeah so that is what asserts are for: explicitly state assumptions in the code, violations of which are caught in debug mode , without introducing crashes in release mode